### PR TITLE
add Composer support

### LINKED ISSUE #52: Composer & Packagist support
Task / Feature request:

Add [composer.json](http://getcomposer.org) and submit the module to [packagist.org](http://packagist.org) - for easier inclusion in Composer-powered Kohana projects.

### PR DESCRIPTION
**Pull request by cmwelsh**
I added Composer support. The package just needs to be registered on [Packagist](https://packagist.org/).

I removed the submodule for Mustache.php and added it as a Composer dependency. I also deleted the autoloader for Mustache.php because this is handled by Composer.

Fixes #52.

### CODE REVIEW

**Comment by zombor** [comment]
I'm not entirely sure I like basically requiring composer. These changes basically destroy the ability for anyone to use a submodule for this.

**Comment by cmwelsh** [comment]
I understand. I'm still learning the Kohana way and don't want to break CFS. I'll revoke this pull request for now.

**Comment by zombor** [comment]
If we can figure out a way to support both, I'd gladly merge this :)

**Comment by cmwelsh** [comment]
This should satisfy your issues with the last patch. I haven't done extensive testing but I wanted to get this out there for you in case you are interested. Thanks and I'll let you know if I get some time to extensively test this approach. This seems like the best way to improve Kohana module packaging - I hate Git submodules.

**Comment by zombor** [comment]
There's a composer kohana installer thing here: https://github.com/composer/installers/blob/master/src/Composer/Installers/KohanaInstaller.php

If you put `'type' => 'kohana-module'` I think that will do it. I'm not entirely happy with this, as it doesn't work if you don't have your modules in the "stock" location, but it's something.

**Comment by cmwelsh** [comment]
I think you're right about the custom installer. If users want to put their Kohana modules in a non-stock location, it looks pretty easy to fork the KohanaInstaller package and change the default settings.

http://getcomposer.org/doc/articles/custom-installers.md

**Comment by cmwelsh** [comment]
I have pushed out a new patch. I think this is the way to go for Composer support of Kohana modules. It's quite easy to add custom sources to your project so I'm already using Kostache as a Composer package. See my `composer.json` here:

https://raw.github.com/cmwelsh/kohana-projecttemplate/3.3/master/composer.json

What I'm saying is that if you want to override the default Composer Installer location for Kohana modules, you can just fork the `composer/installers` package where `KohanaInstaller` lives and add your fork to the `repositories` section of your `composer.json`.

I might do a quick blog post about this so I can share with my coworkers.

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+	"name": "kohana-modules/kostache",
```
Comment (cs278) [inline composer.json]: Should be `zombor/kostache` per http://getcomposer.org/doc/04-schema.md#name

Comment (cmwelsh) [inline composer.json]: Yeah, you're right. I'm still a bit of a novice with Composer.

**Comment by happyDemon** [comment]
+1 for this last  pull request, shouldn't break anything, can this be merged?

**Comment by evanpurkhiser** [comment]
:+1: for this, any chance of seeing this merged?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,3 @@
 *.swp
+/vendor/
```
Comment (shadowhand) [inline .gitignore:2]: wouldn't this break the submodule?

Comment (zombor) [inline .gitignore:2]: Yep, https://github.com/zombor/KOstache/pull/58#issuecomment-12737359

**Comment by evanpurkhiser** [comment]
Any chance of reconsidering this since it seems like other kohana modules are moving towards composer only (i.e. [shadowhand/email](https://github.com/shadowhand/email) and [zeelot/kohana-media](https://github.com/Zeelot/kohana-media))?

File: init.php [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <?php defined('SYSPATH') or die('No direct script access.');
 
 // Load Mustache for PHP
-include Kohana::find_file('vendor', 'mustache/src/Mustache/Autoloader');
-Mustache_Autoloader::register();
+if ( ! class_exists('Mustache_Engine')) {
```
Comment (shadowhand) [inline init.php:4]: this is not necessary, the composer autoloader will take care of it.

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+	"name": "zombor/kostache",
+	"type": "kohana-module",
```
Comment (shadowhand) [inline composer.json:3]: we are moving away from `kohana-module` type packages, especially for 3rd party modules, see [this comment](https://github.com/Zeelot/kohana-media/issues/8#issuecomment-36151728).

File: .gitignore [hunk 5]
```diff
@@ -1 +1,3 @@
 *.swp
```
Comment (shadowhand) [inline .gitignore:1]: this entire file should be deleted, .gitignore does not belong in modules.

**Comment by shadowhand** [comment]
I will merge this, if it fixes the issues mentioned above.

**Comment by shadowhand** [comment]
Fixed by #69.

**Pull request closed without merging**